relay-server/src/actors/events.rs

```diff
@@ -1696,11 +1700,25 @@ impl Handler<HandleEnvelope> for EventManager {
                     outcome_producer.do_send(TrackOutcome {
                         timestamp: Instant::now(),
                         scoping: *scoping.borrow(),
-                        outcome,
+                        outcome: outcome.clone(),
                         event_id,
                         remote_addr,
                         category,
-                    })
+                        quantity: 1,
+                    });
+
+                    let envelope_summary = envelope_summary.borrow();
+                    if envelope_summary.attachment_quantity > 0 {
```

Comment:
This actually needs to move, and I believe you'll want to restructure this error handler at this point. A couple of lines above, there is a check that's no longer accurate:

```rust
// Envelopes not containing events (such as standalone attachment uploads or user
// reports) should never create outcomes.
let category = match event_category {
    Some(event_category) => event_category,
    None => return,
};
```

Instead, we now need to emit outcomes for attachments, just not for the event. That would be similar to `endpoints/common.rs`.

Please also pay attention to another early return condition, which still holds:

```rust
// Do not emit outcomes for requests that have been accepted by the upstream. In
// such a case, the upstream assumes ownership and logs the outcome, instead. This
// is irrelevant to processing Relays, since they never send to the upstream.
if is_received.load(Ordering::SeqCst) {
    return;
}
```